thrust/detail/reference.h

```diff
@@ -141,24 +141,9 @@ template<typename Element, typename Pointer, typename Derived>
     __host__ __device__
     inline void assign_from(OtherPointer src);
 
-    // XXX this helper exists only to avoid warnings about null references from the other assign_from
-    template<typename System1, typename System2, typename OtherPointer>
-    inline __host__ __device__
-    void assign_from(System1 *system1, System2 *system2, OtherPointer src);
-
     template<typename System, typename OtherPointer>
     __host__ __device__
     inline void strip_const_assign_value(const System &system, OtherPointer src);
```

Comment:
Is this `strip_const_assign_value()` function superfluous? It seems like code that is calling it should just be able to call `assign_value()` directly.